rust/otel-arrow-rust/benches/item_count.rs

```diff
@@ -0,0 +1,104 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(missing_docs)]
+
+//! This crate benchmarks counting log records in an OTLP LogsData.
+
+use criterion::{Criterion, black_box, criterion_group, criterion_main};
+
+use otel_arrow_rust::pdata::otlp::ItemCounter;
+use otel_arrow_rust::pdata::otlp::LogsVisitor;
+
+use otel_arrow_rust::proto::opentelemetry::common::v1::*;
+use otel_arrow_rust::proto::opentelemetry::logs::v1::*;
+use otel_arrow_rust::proto::opentelemetry::resource::v1::*;
+
+fn create_logs_data() -> LogsData {
```

Comment:
It’s too early for now, but one of these days we should create a shared crate for fake OTLP data generation in `otap-dataflow`. There’s this code base, and there’s another example here: https://github.com/open-telemetry/otel-arrow/blob/main/rust/otap-dataflow/crates/otap/src/parquet_exporter/test/datagen.rs